mkn.yaml

```diff
@@ -0,0 +1,167 @@
+               -Wl,-z,relro -Wl,-z,relro
+    bsd_largs: ${nixish_largs} 
+
+    nixish_test_cargs: -Wno-error=unused-local-typedefs 
+                       -Wno-error=sign-compare -Wno-sign-compare
+
+profile:
+  - name: arg
+    arg: ${cargs} 
+    if_arg:
+      win: -GL -MD -EHsc -Zc:inline
+      nix: ${nixish_cargs}
+      bsd: ${nixish_cargs}
+      win_shared: -DBUILDING_DLL
+    if_link:
+      nix_lib: ${nix_largs}
+      bsd_lib: ${bsd_largs}
+
+  - name: base
+    parent: arg
+    self: array
+    src:
+      ./tick/base/src
+      ./tick/base/swig
+    install: ${lib_path}/base/build
+    out: base${lib_name}
+
+  - name: array
+    parent: arg
+    inc: 
+      ./tick/base/src 
+      ./tick/base/array/src
+      ./tick/third_party/cereal/include
+    src:
+      ./tick/base/array/src
+      ./tick/base/array/swig
+    install: ${lib_path}/base/array/build
+    out: array${lib_name}
+
+  - name: array_test
+    parent: arg
+    self: base
+    inc: 
+      ./tick/base/array_test/src
+    src:
+      ./tick/base/array_test/src
+      ./tick/base/array_test/swig
+    install: ${lib_path}/base/array_test/build
+    out: array_test${lib_name}
+
+  - name: inference
+    parent: arg
+    self: optim/model
+    inc: ./tick/inference/src
+    src:
+      ./tick/inference/src
+      ./tick/inference/swig
+    install: ${lib_path}/inference/build
+    out: inference${lib_name}
+
+  - name: optim/model
+    parent: arg
+    self: base
+    inc: ./tick/optim/model/src
+    src:
+      ./tick/optim/model/src
+      ./tick/optim/model/swig
+    install: ${lib_path}/optim/model/build
+    out: model${lib_name}
+
+  - name: optim/prox
+    parent: arg
+    self: base
+    inc: ./tick/optim/prox/src
+    src:
+      ./tick/optim/prox/src
+      ./tick/optim/prox/swig
+    install: ${lib_path}/optim/prox/build
+    out: prox${lib_name}
+
+  - name: optim/solver
+    parent: arg
+    self: random optim/model optim/prox
+    inc: ./tick/optim/solver/src
+    src:
+      ./tick/optim/solver/src
+      ./tick/optim/solver/swig
+    install: ${lib_path}/optim/solver/build
+    out: solver${lib_name}
+
+  - name: preprocessing
+    parent: arg
+    self: base
+    inc: ./tick/preprocessing/src
+    src:
+      ./tick/preprocessing/src
+      ./tick/preprocessing/swig
+    install: ${lib_path}/preprocessing/build
+    out: preprocessing${lib_name}
+
+  - name: simulation
+    parent: arg
+    self: random
+    inc: ./tick/simulation/src
+    src:
+      ./tick/simulation/src
+      ./tick/simulation/swig
+    install: ${lib_path}/simulation/build
+    out: simulation${lib_name}
+
+  - name: random
+    parent: arg
+    self: base
+    inc: ./tick/random/src
+    src:
+      ./tick/random/src
+      ./tick/random/swig
+    install: ${lib_path}/random/build
+    out: crandom${lib_name}
+
+### See file ./sh/gtest/sh 
+  - name: gtest
+    parent: arg
+    main: tick/optim/model/tests/src/hawkes_models_gtest.cpp
```

Comment:
Wouldn't this only launch one test file?

Reply:
Should it be kept since `sh/gtest.sh` exists?

Reply:
It's required by "sh/gtest.sh".

Reply:
Oh ok, so this is a default parameter. Maybe should we give example on how to use each .sh file?

Reply:
It's explained at the top of the gtest.sh file - how it works, and how to use it - let me know if it's ok

Reply:
Ok sorry, I'll give a more careful look when I will install it on my other machine